### DISCUSSION TITLE
Is it possible to control (e.g. reverse) the nesting hierarchy?

### DISCUSSION THREAD

**Discussion by lwestenberg** [facebook/lexical#4262] [Q&A]
When not using registerNestedElementResolver:
- I make a text selection
- Create a node from that text selection, e.g. MarkNode
- Within that MarkNode I make a text selection
- I create another node, e.g. CustomMarkNode

the nodes are now nested in the order of adding e.g. in the above example it would become:
`<Marknode>
  <CustomMarkNode>
  </CustomMarkNode>
</MarkNode>`

Is it possible to control which node becomes the parent and which one becomes the child? In my situation I want to have a specific node type always as the parent.

**Comment by moy2010** [comment]
Yes. Therer are methods such as `isParentRequired` and `createParentElementNode` on the `LexicalNode` class which you can leverage to achieve the behaviour you want:

```
/**
   * Whether or not this node has a required parent. Used during copy + paste operations
   * to normalize nodes that would otherwise be orphaned. For example, ListItemNodes without
   * a ListNode parent or TextNodes with a ParagraphNode parent.
   *
   * */
  isParentRequired(): boolean {
    return false;
  }
  
  /**
   * The creation logic for any required parent. Should be implemented if {@link isParentRequired} returns true.
   *
   * */
  createParentElementNode(): ElementNode {
    return $createParagraphNode();
  }
```
